### DISCUSSION TITLE
컴퓨터 네트워크(민수-> 승은)

### DISCUSSION THREAD

**Discussion by skyriv213** [tempdku/temp#5] [General]
컴퓨터 네트워크 질의 응답
@sep037

**Comment by sep037** [comment]
덤벼

**Reply by skyriv213** [reply]
10,11번
HTTPS와의 차이점을 적었는데 HTTPS는 어떤 방식을 통해 HTTP에서 보안을 강화한건지? 
그리고 이때 데이터의 처리는 어떻게 진행이 되는지? 
SSL과 TLS는 어떤 방식을 추가하여 개선이 진행되었는지

17번
해당 항목에 대해 더 자세히 설명 ㄱㄱㄱ ^^

**Reply by sep037** [reply]
10,11
----
- HTTP는 전송 중인 데이터를 가로채면 누구나 데이터를 읽을 수 있어요. HTTPS는 이런 보안적 문제를 해결한 프로토콜입니당. HTTPS를 SSL 프로토콜 위에서 돌아가도록 하여 클라이언트와 서버가 주고 받는 텍스트를 암호화 하는 기법을 이용하는데요, 즉 HTTPS = HTTP + SSL 프로토콜인 것 같습니다. (대칭키 암호화, 비대칭키 암호화) 
- SSL/TLS는 텍스트를 아무나 읽지 못하도록 인코딩하는 알고리즘입니다. 응용계층과 전송계층 사이에 위치하고 있으며 키에 따라 암호화 결과가 달라지기 때문에 키를 모르면 복호화가 불가능합니다. 

17
----
1. 사용자는 url을 웹 브라우저에 입력합니다.
2. 웹 브라우저는 입력한 url을 바탕으로 DNS 서버에 IP 요청을 보냅니다.
3. DNS는 IP 주소를 웹 브라우저에 응답을 제공합니다.
4. 웹 브라우저는 DNS 서버에서 받은 IP를 통해 웹 서버와 TCP/IP 연결을 하고 HTTP 요청을 보냅니다.
5. 웹 서버는 받은 HTTP 요청에 응답하고, 응답은 웹 페이지와 필요한 리소스를 포함합니다.
6. 웹 브라우저는 받은 응답을 바탕으로 사용자에게 웹 페이지를 보여줍니다.

끗

**Reply by skyriv213** [reply]
그럼 혹시 SSL과 TLS가 무슨 차이점이 존재하는지도 말해주실 수 있나요??

**Reply by sep037** [reply]
TLS는 SSL보다 더 강력한 암호화 알고리즘을 사용함 
SSL은 오래됨
-> 결론 : TLS써라
